systest/tests/distributed_post_verification_test.go

```diff
@@ -0,0 +1,312 @@
+	nodeID := signer.NodeID()
+	atx.InnerActivationTx.NodeID = &nodeID
+	require.NoError(t, activation.SignAndFinalizeAtx(signer, atx))
+
+	// 3. Wait for publish epoch
+	epoch := atx.PublishEpoch
+	logger.Sugar().Infow("waiting for publish epoch", "epoch", epoch, "layer", epoch.FirstLayer())
+	err = layersStream(ctx, cl.Client(0), logger, func(resp *pb.LayerStreamResponse) (bool, error) {
+		logger.Info("new layer", zap.Uint32("layer", resp.Layer.Number.Number))
+		return resp.Layer.Number.Number < epoch.FirstLayer().Uint32(), nil
+	})
+	require.NoError(t, err)
+
+	// 4. Publish ATX
+	publishCtx, stopPublishing := context.WithCancel(ctx.Context)
+	t.Cleanup(stopPublishing)
+	var eg errgroup.Group
+	eg.Go(func() error {
+		for {
+			logger.Sugar().Infow("publishing ATX", "atx", atx)
+			buf, err := codec.Encode(atx)
+			require.NoError(t, err)
+			err = host.Publish(ctx, pubsub.AtxProtocol, buf)
+			require.NoError(t, err)
+
+			select {
+			case <-publishCtx.Done():
+				return nil
+			case <-time.After(10 * time.Second):
+			}
+		}
+	})
+
+	// 5. Wait for POST malfeasance proof
+	logger.Info("waiting for malfeasance proof")
+	err = malfeasanceStream(ctx, cl.Client(0), logger, func(malfeasance *pb.MalfeasanceStreamResponse) (bool, error) {
+		stopPublishing()
+		logger.Info("malfeasance proof received")
+		require.Equal(t, malfeasance.GetProof().GetSmesherId().Id, signer.NodeID().Bytes())
+		require.Equal(t, pb.MalfeasanceProof_MALFEASANCE_POST_INDEX, malfeasance.GetProof().GetKind())
+
+		var proof types.MalfeasanceProof
+		require.NoError(t, codec.Decode(malfeasance.Proof.Proof, &proof))
+		require.Equal(t, types.InvalidPostIndex, proof.Proof.Type)
+		invalidPostProof := proof.Proof.Data.(*types.InvalidPostIndexProof)
+		logger.Sugar().Infow("malfeasance post proof", "proof", invalidPostProof)
+		invalidAtx := invalidPostProof.Atx
+		require.Equal(t, atx.PublishEpoch, invalidAtx.PublishEpoch)
+		require.Equal(t, atx.SmesherID, invalidAtx.SmesherID)
+		require.Equal(t, atx.NodeID, invalidAtx.NodeID)
+		require.Equal(t, atx.PositioningATX, invalidAtx.PositioningATX)
+		require.Equal(t, atx.PrevATXID, invalidAtx.PrevATXID)
+		require.Equal(t, atx.Signature, invalidAtx.Signature)
+		require.Equal(t, atx.Coinbase, invalidAtx.Coinbase)
+		require.Equal(t, *atx.CommitmentATX, *invalidAtx.CommitmentATX)
+		require.Equal(t, atx.NIPostChallenge, invalidAtx.NIPostChallenge)
+		require.Equal(t, atx.NIPost.Post.Indices, invalidAtx.NIPost.Post.Indices)
+
+		postVerifier, err := activation.NewPostVerifier(cfg.POST, logger.Named("post-verifier"))
+		require.NoError(t, err)
+		meta := &shared.ProofMetadata{
+			NodeId:          invalidAtx.NodeID.Bytes(),
+			CommitmentAtxId: invalidAtx.CommitmentATX.Bytes(),
+			NumUnits:        invalidAtx.NumUnits,
+			Challenge:       invalidAtx.NIPost.PostMetadata.Challenge,
+			LabelsPerUnit:   invalidAtx.NIPost.PostMetadata.LabelsPerUnit,
+		}
+		err = postVerifier.Verify(ctx, (*shared.Proof)(invalidAtx.NIPost.Post), meta)
```

Comment:
NIT: you already have an instance of the verifier further up.

```suggestion
		verifier.Verify(ctx, (*shared.Proof)(invalidAtx.NIPost.Post), meta)
```